modules/coredns/collect.go

```diff
@@ -89,13 +81,34 @@ func (cd *CoreDNS) collect() (map[string]int64, error) {
 	return stm.ToMap(mx), nil
 }
 
-// Summary
+func (cd *CoreDNS) updateVersionDependentMetrics(raw prometheus.Metrics) {
+	if cd.version == nil {
+		if version := cd.Version(raw); version != nil {
```

Comment:
Perhaps minor, but still. In case of a version parse error, we can assume that it doesn't make sense to keep trying it - otherwise we will log parse error on every data collection.

Reply:
I've added some check to avoid retrying. I also had to change some receiver functions from values to pointer because we were passing copies of the `CoreDNS` objects and state changes were lost.
https://github.com/netdata/go.d.plugin/pull/619/files#diff-425bfb25babae1534ae9cac4c982308ab535f71e214b4661abc052a063adfc3dR109
and 
https://github.com/netdata/go.d.plugin/pull/619/files#diff-425bfb25babae1534ae9cac4c982308ab535f71e214b4661abc052a063adfc3dR114

Reply:
Please be careful, I am sure if there if they were value receivers for some specific reason.